x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/planner/QueryTranslatorTests.java

```diff
@@ -20,30 +20,40 @@
 import org.elasticsearch.xpack.sql.planner.QueryTranslator.QueryTranslation;
 import org.elasticsearch.xpack.sql.querydsl.query.Query;
 import org.elasticsearch.xpack.sql.querydsl.query.RangeQuery;
+import org.elasticsearch.xpack.sql.querydsl.query.ScriptQuery;
 import org.elasticsearch.xpack.sql.querydsl.query.TermQuery;
+import org.elasticsearch.xpack.sql.querydsl.query.TermsQuery;
 import org.elasticsearch.xpack.sql.type.EsField;
 import org.elasticsearch.xpack.sql.type.TypesTests;
 import org.joda.time.DateTime;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
 
+import java.io.IOException;
 import java.util.Map;
 import java.util.TimeZone;
 
-public class QueryTranslatorTests extends ESTestCase {
+import static org.hamcrest.core.StringStartsWith.startsWith;
 
-    private SqlParser parser;
-    private IndexResolution getIndexResult;
-    private FunctionRegistry functionRegistry;
-    private Analyzer analyzer;
-    
-    public QueryTranslatorTests() {
+public class QueryTranslatorTests extends AbstractBuilderTestCase {
```

Comment:
Is `AbstractBuilderTestCase` used anywhere (maybe I'm missing it)?

Reply:
I used it for the shardContext here: https://github.com/elastic/elasticsearch/pull/34581/files/ab7c1502d1b6341b2b3cc3be7eda72b00e606ada#diff-aef2b0ce456b8fdd5cc09d6cfd55f0c2R173.
I tried to manually mock it but ended up with ugly code. The AbstractBuilderTestCase is parent class for many tests in `.search.aggregations` and `.index.query` packages.